src/pattern-library/components/Library.tsx

```diff
@@ -179,32 +199,94 @@ export type LibraryDemoProps = {
   withSource?: boolean;
 };
 
+/**
+ * Fetches provided example file and returns its contents as text, excluding
+ * the import statements.
+ * An error is thrown if the file cannot be fetched for any reason.
+ */
+async function fetchCodeExample(
+  exampleFile: string,
+  signal: AbortSignal,
+): Promise<string> {
+  const res = await fetch(`/examples/${exampleFile}.tsx`, { signal });
+  if (res.status >= 400) {
+    throw new Error(`Failed loading ${exampleFile} example file`);
+  }
+
+  const text = await res.text();
+
+  // Remove import statements and trim trailing empty lines
+  return text.replace(/^import .*;\n/gm, '').replace(/^\s*\n*/, '');
+}
+
+function useDemoChildrenAndSource(
+  props: Pick<LibraryDemoProps, 'children' | 'exampleFile'>,
+) {
+  const [source, setSource] = useState<ComponentChildren>();
+  const [children, setChildren] = useState<ComponentChildren>();
```

Comment:
In the case where the source and children are directly available, the values could be computed and used to initialize the returned state synchronously.

Reply:
Hmm, that's a good idea.